app/src/test/kotlin/io/intrepid/skotlinton/screens/example2/Example2PresenterTest.kt

```diff
@@ -9,9 +9,9 @@ import org.mockito.Mock
 import org.mockito.Mockito.`when`
 import org.mockito.Mockito.verify
 
-internal class Example2PresenterTest : BasePresenterTest<Example2Presenter>() {
+internal class Example2PresenterTest : BasePresenterTest<Example2Screen, Example2Presenter>() {
```

Comment:
any reason for including the screen class as part of generics? Ideally the mockView class would live in the base class, but the limitation of Mockito's Mock annotation prevents that

Reply:
Yes I tried to do this and noticed mockito doesn't work well with generics. I believe I was having trouble making `BaseActivity` and `BaseFragment` work by taking in only a generic child `BasePresenter<BaseScreen>`

Reply:
Maybe we should move to [`mockito-kotlin`](https://github.com/nhaarman/mockito-kotlin). It has some reified generic methods that could help, though I haven't played with this specific use case.

Reply:
This is more of a issue with Mockito than Kotlin, it doesn't work in Java either